src/installer/installer.go

```diff
@@ -466,37 +462,38 @@ func (i *installer) waitForController() error {
 		i.log.WithError(err).Errorf("Failed to create kc client from %s", KubeconfigPath)
 		return err
 	}
+
 	events := map[string]string{}
-	predicate := func() bool {
-		controllerPod := common.GetPodInStatus(kc, assistedControllerPrefix, assistedControllerNamespace,
-			map[string]string{"job-name": assistedControllerPrefix}, v1.PodRunning, i.log)
-		if controllerPod != nil {
-
-			// uploading logs here cause we will finish installing bootstrap in couple of seconds
-			// just didn't wanted to start handling this logic in other place
-			// controller must be running for at least couple of minutes before this code so it will cover 90% of the cases
-			// if we will see that it is not enough, we will change the place.
-			err = common.UploadPodLogs(kc, i.inventoryClient, i.ClusterID, controllerPod.Name, assistedControllerNamespace, common.ControllerLogsSecondsAgo, i.log)
-			// if failed to upload logs, log why and continue
-			if err != nil {
-				if retryCounterUploadControllerLogs > 0 {
-					retryCounterUploadControllerLogs--
-					controllerPod = nil
+	tickerUploadLogs := time.NewTicker(5 * time.Minute)
+	tickerWaitForController := time.NewTicker(waitForControllerPodInterval)
+	// Keep trying until we're time out or get true
+	for {
+		select {
+		case <-tickerWaitForController.C:
```

Comment:
You should use a watcher/informer here instead of the ticker.
https://pkg.go.dev/k8s.io/client-go/informers#SharedInformerFactory

Reply:
as we talked in this case there is no reason to implement informer logic, this code in 99% will run only once